doc/installation-guide.md

```diff
@@ -126,3 +122,35 @@ colcon build
 ```
 
 This build method makes the most sense in combination with [rmw_iceoryx](https://github.com/ros2/rmw_iceoryx.git)
+
+## Build and run tests
+
+While developing on iceoryx you want to know if your changes are breaking existing functions or if your newly written googletests are passing.
+For that purpose we are generating cmake targets which are executing the tests. But first we need to build them:
+```
+cmake -Bbuild -Hiceoryx_meta -DBUILD_TEST=ON
+cmake --build build
+```
+Cmake is automatically installing googletest as dependency and build the tests against it. Please note that if you want to build tests for extensions like the C-Binding you need to enable that in the cmake build. To build all tests simply add `-DBUILD_ALL` to the cmake command
+
+Now lets execute the all tests:
+```
+cd iceoryx/build
+make all_tests
+```
+Some of the tests are timing critical and needs a stable environment. We call them timing tests and have them in a separate target available:
+```
+make timing_tests
+```
+In iceoryx we distinguish between different testlevels. The most important are: Moduletests and Integrationtests.
+Moduletests are basically Unit-tests where the focus is on class level with black-box testing.
+In Integrationtests are multiple classes within one component (e.g. iceoryx_posh) tested together.
+The sourcecode of the tests is placed into the folder `test` within the different iceoryx components. You can find there at least a folder `moduletests` and sometimes ``integrationtests`.
+
+when you now want to create a new test you can place the sourcefile directly into the right folder. Cmake will automatically detect the new file when doing a clean build and will add it to a executable. There is no need to add a gtest main function because we already provide it.
+In every are then the executables for every test level created, for example `posh_moduletests`. They are placed into the corresponding build folder (e.g. `iceoryx/build/posh/test/posh_moduletests`).
+
+If you want to execute only individual testcases then you can use these executables and a gtest filter. Let's assume you want to execute only the `ServiceDescription_test` from the posh_moduletests, then you can do the following:
+```
+./build/posh/test/posh_moduletests --gtest_filter=ServiceDescription_test*
```

Comment:
Depending on the shell this may not work but when adding some double quotes like here 
`./build/posh/test/posh_moduletests --gtest_filter="ServiceDescription_test*"` it will work everywhere.